deltacat/compute/compactor/steps/materialize.py

```diff
@@ -154,8 +149,7 @@ def _materialize(
             src_file_idx_np.item(),
             file_reader_kwargs_provider=read_kwargs_provider,
         )
-        record_count = len(pa_table)
-        mask_pylist = list(repeat(False, record_count))
+        mask_pylist = list(repeat(False, len(pa_table)))
```

Comment:
We don't need to create a mask pylist after my latest changes. We could directly pass pyarrow array here to `take` indices that we need.